merkle/src/tree.rs

```diff
@@ -14,9 +16,39 @@ pub trait MerkleTree<const DEGREE: usize> {
     fn count(&self) -> usize;
     /// Root hash, lazy-evaluated
     fn root(&mut self) -> Option<Self::Hash>;
-
     /// Adds a new leaf
-    fn add(&mut self, item: Self::Item);
+    fn add(&mut self, item: Self::Item) {
+        self.add_batch(vec![item])
+    }
     /// Adds a batch of leaves in the give order
     fn add_batch(&mut self, items: impl IntoIterator<Item = Self::Item>);
 }
+
+/// Trait for implementing sparse binary merkle tree.
+/// It is essentially a collection of [MerkleProof]
+pub trait SparseMerkleTree {
+    /// Type of the leaf data
+    type Item;
+    /// Type of the hash values
+    type Hash: PartialEq + Clone;
+    /// Type of the merkle hasher
+    type Hasher: MerkleHasher<Item = Self::Item, Hash = Self::Hash>;
+    /// Type of the merkle merger
+    type Merger: MerkleMerger<Hash = Self::Hash>;
+
+    /// Adds a single [MerkleProof]
+    fn add(
```

Comment:
Make this generic over types that implement `MerkleProof` rather than forcing it to be a `DefaultMerkleProof`

Reply:
Good catch! Fixed